### ISSUE TITLE
Lightdm + hostname + dhcp name issue

### ISSUE THREAD

**Issue by chrish13** [canonical/lightdm#376]
Not sure about my title but here is the issue that we face at my work.
We have a few dozens of NUC pcs that are cloned from a master image. We do not want them with static IP address, or static name.
For that we have configured the NetworkManger to be DHCP agnostic and even fetch a hostname from the DHCP server. Truncate the file `/etc/hostname` with the command `# truncate -s 0 /etc/hostname` 
Our DHCP servers are passing them all the details that the clients need including their hostname. 
So when a new clone boots LightDM shows 'localhost.localdomain' as info (top left widget). When an ldap user or local user tries to login we get just a blank (or better black) screen with just the cursor. Switching to VT, for example VT 1 (ctrl+alt+f1) we get the prompt:
`localhost login:`
We can now login as any user, no problem at all. The prompt now changes to the logged in user with the dhcp name ( for example `user1@dev-201:~$` ) The 'dev-201' is given by the dhcp, which is automatic depending the IP 's ending.
However, login in as root and restart the lightdm service: `systemctl restart lightdm.service` and returning back to our VT7 (default gui terminal), after login in again successfully we do have a desktop.
To temporary resolve the issue we have set the hostname to localhost but this is not we would like to have.
We haven't test this on any other Debian/Ubuntu flavor.

Client OS: Linux Mint 6 Debian edition

Unfortunately by looking any log file did not shed any light.
Xorg logs says everything is alright, same for lightdm, sssd, pam.

To summarize:
Blank /etc/hostname
No desktop after successful login (local or ldap)

Valid entry on /etc/hostname:
Successful desktop access after successful login.
